packages/app-page-builder/src/blockEditor/config/editorBar/SaveBlockButton/SaveBlockButton.tsx

```diff
@@ -1,32 +1,55 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useState } from "react";
+import styled from "@emotion/styled";
 import { createComponentPlugin, makeComposable } from "@webiny/app-admin";
 import { useSnackbar } from "@webiny/app-admin/hooks/useSnackbar";
 import { useRouter } from "@webiny/react-router";
 import { ButtonPrimary } from "@webiny/ui/Button";
+import { CircularProgress } from "@webiny/ui/Progress";
 import { EditorBar } from "~/editor";
 import { useEventActionHandler } from "~/editor/hooks/useEventActionHandler";
 import { useBlock } from "~/blockEditor/hooks/useBlock";
 import { SaveBlockActionEvent } from "~/blockEditor/config/eventActions/saveBlock/event";
 
+const LoadingWrapper = styled("div")({
+    position: "absolute",
+    width: "100vw",
+    height: "100vh",
+    top: 0,
+    right: 0,
+    color: "var(--mdc-theme-text-primary-on-background)"
+});
+
 const DefaultSaveBlockButton: React.FC = () => {
     const [block] = useBlock();
     const eventActionHandler = useEventActionHandler();
     const { history } = useRouter();
     const { showSnackbar } = useSnackbar();
+    const [loading, setLoading] = useState(false);
 
     const saveChanges = useCallback(() => {
+        setLoading(true);
         eventActionHandler.trigger(
             new SaveBlockActionEvent({
                 debounce: false,
                 onFinish: () => {
+                    setLoading(false);
                     history.push(`/page-builder/page-blocks`);
                     showSnackbar(`Block "${block.name}" saved successfully!`);
                 }
             })
         );
     }, [block.name]);
 
-    return <ButtonPrimary onClick={saveChanges}>Save Changes</ButtonPrimary>;
+    return (
+        <>
+            {loading && (
+                <LoadingWrapper>
+                    <CircularProgress label={"Saving block..."} />
+                </LoadingWrapper>
+            )}
+            <ButtonPrimary onClick={saveChanges}>Save Changes</ButtonPrimary>
+        </>
+    );
```

Comment:
Can we make this loading indicator local to the button only? The overlay looks a little over the top. We can just add a spinner to the button, and disable the button while the saving is taking place.

The code for button with icon goes like this:
```jsx
<ButtonPrimary>
  <ButtonIcon icon={<SvgBaselineCloudDone24Px />} />
  Click to proceed
</ButtonPrimary>
```

Reply:
@Pavel910 just in case: I am adding smth like this, right?
![image](https://user-images.githubusercontent.com/77202393/204580012-e932872e-17fc-44a7-a239-a84321db9881.png)

Reply:
@SvenAlHamad @Pavel910 I want to confirm one thing:
We have such "saved"/"saving" marker in the Page Editor (it is showing save progress after each edit action - revision is updated when done).

On Block Editor we have same icon - but it is not used (as we save block only on "save" button click).
Should I remove it from Block Editor? or use it to show the block save progress (keep in mind that we already have a spinner inside the button)?

Other thing I want to confirm is about how to determine an unsaved changes.
1) we can track edit actions (if edits >0 - smth is unsaved)
(if there were edits, but we undo all of them - block is marked as changed, as edits > 0)
2) we can compare current block state with initial state
(if there were edits, but we undo all of them - block is marked as NOT changes, as initial and current block objects are same)


![image](https://user-images.githubusercontent.com/77202393/204779377-1e13bb1e-6dee-41d8-8bf3-f3e7f78c6cdd.png)

Reply:
Regarding the save icon - let's remove that one from the left sidebar as we don't have autosave functionality inside the block editor.

Reply:
@neatbyte-vnobis The indicator kicks in when the actual API call is made, and it does work, so just leave it there.
Maybe you can hook into the `UpdateElementActionEvent`, and whenever it's triggered, you store your own state, which means the block is "dirty". Then you also hook into the `SaveBlockActionEvent`, and when it's executed, you can update your state as "not dirty".

Essentially you just listen for those 2 events and track some state of your own, a dedicated recoil atom, or just simple React statel

Reply:
@neatbyte-vnobis about that indicator again... I ignored the tooltip previously; yeah it makes no sense for the Block editor. What we could do is replace that widget with a Block Editor specific one, which will show like an exclamation mark when the block is not saved 🤔

Reply:
@neatbyte-vnobis sorry, sven posted too quick, and I missed his message.
So yes, remove the icon.

About the undo/redo, when those events happen, we also execute two event actions: `UpdateDocumentActionEvent` and `UpdateElementTreeActionEvent`, so you could also hook into that, and track the "dirty" state that way. So regardless of the action, undo or redo, just track it as "dirty".